src/main/java/mServer/crawler/sender/orf/tasks/AbstractDocumentTask.java

```diff
@@ -0,0 +1,59 @@
+package mServer.crawler.sender.orf.tasks;
+
+import de.mediathekview.mlib.tool.Log;
+import java.io.IOException;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import mServer.crawler.sender.MediathekReader;
+import mServer.crawler.sender.orf.CrawlerUrlDTO;
+import org.jsoup.HttpStatusException;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+
+/**
+ * This is a abstract task based on {@link AbstractUrlTask} which takes a
+ * {@link ConcurrentLinkedQueue} of {@link D} and loads the URL with JSOUP as
+ * {@link Document}.
+ *
+ * @param <T> The type of objects which will be created from this task.
+ * @param <D> A sub type of {@link CrawlerUrlDTO} which this task will use to
+ * create the result objects.
+ */
+public abstract class AbstractDocumentTask<T, D extends CrawlerUrlDTO>
+        extends AbstractUrlTask<T, D> {
+
+  private static final long serialVersionUID = -4124779055395250981L;
+  private static final String LOAD_DOCUMENT_HTTPERROR
+          = "Some HTTP error happened while crawl the %s page \"%s\".";
+
+  public AbstractDocumentTask(final MediathekReader aCrawler,
+          final ConcurrentLinkedQueue<D> aUrlToCrawlDTOs) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The type of the "aUrlToCrawlDTOs" object should be an interface such as "Queue" rather than the implementation "ConcurrentLinkedQueue". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1319)